base_layer/core/src/blocks/block.rs

```diff
@@ -197,7 +197,7 @@ impl BlockBuilder {
     /// This function adds the provided transaction kernels to the block WITHOUT updating kernel_mmr_size in the header
     pub fn add_kernels(mut self, mut kernels: Vec<TransactionKernel>) -> Self {
         for kernel in &kernels {
-            self.total_fee += kernel.fee;
+            self.total_fee = self.total_fee.saturating_add(kernel.fee);
```

Comment:
I'm curious what the threshold is for blocks so big the fee reaches a saturated max, but max block size hasn't been reached yet. Making blocks of that size and larger the same recommended fee. Or even if it's possible to end up with a recommended fee per gram large enough to reach the max.

Reply:
I think that is the crux of the matter; it is possible to end up with such a large fee just with a few transactions

Reply:
I added a test for this case